apps/cow-tools/src/app/milkman/(components)/OrdersTable.tsx

```diff
@@ -53,38 +60,211 @@ export function OrderTable({ orders }: { orders: AllSwapsQuery["swaps"] }) {
+  transaction,
+  hasToken,
+  cowOrders,
+}: {
+  transaction: AllTransactionFromUserQuery["users"][0]["transactions"][0];
+  hasToken?: boolean[];
+  cowOrders?: ICowOrder[][];
+}) {
+  function getSwapStatus(hasToken?: boolean, cowOrders?: ICowOrder[]) {
+    if (!cowOrders || hasToken === undefined) {
+      return SwapStatus.MILKMAN_CREATED;
+    }
+
+    const anyOrderWasExecuted = cowOrders.some(
+      (order) => order.status == "fulfilled",
+    );
+
+    if (anyOrderWasExecuted) {
+      return SwapStatus.EXECUTED;
+    }
+
+    if (!anyOrderWasExecuted && !hasToken) {
+      return SwapStatus.CANCELED;
+    }
+
+    if (cowOrders.length > 0 && hasToken) {
+      return SwapStatus.ORDER_PLACED;
+    }
+
+    return SwapStatus.MILKMAN_CREATED;
+  }
+
+  function getTransactionStatus(swapStatus: SwapStatus[]) {
+    if (swapStatus.every((status) => status === SwapStatus.CANCELED)) {
+      return TransactionStatus.CANCELED;
+    }
+    if (swapStatus.every((status) => status === SwapStatus.EXECUTED)) {
+      return TransactionStatus.EXECUTED;
+    }
+
+    if (
+      swapStatus.every(
+        (status) =>
+          status === SwapStatus.CANCELED || status === SwapStatus.EXECUTED,
+      )
+    ) {
+      return TransactionStatus.EXECUTED_AND_CANCELED;
+    }
+    if (swapStatus.some((status) => status === SwapStatus.EXECUTED)) {
+      return TransactionStatus.PARTIALLY_EXECUTED;
+    }
+
+    const allOrdersPlaced = swapStatus.every(
+      (status) => status === SwapStatus.ORDER_PLACED,
+    );
+
+    if (allOrdersPlaced) {
+      return TransactionStatus.ORDER_PLACED;
+    }
+
+    return TransactionStatus.MILKMAN_CREATED;
+  }
+
+  const swapStatus = transaction.swaps.map((swap, i) =>
+    getSwapStatus(hasToken?.[i], cowOrders?.[i]),
+  );
+  const transactionStatus = getTransactionStatus(swapStatus);
   const txUrl = buildBlockExplorerTxUrl({
-    chainId: order.chainId,
-    txHash: order.transactionHash,
+    chainId: transaction.swaps[0].chainId,
+    txHash: transaction.id,
   });
 
+  const [showOrdersRows, setShowOrdersRows] = useState(false);
+
+  const equalTokensIn = transaction.swaps.every(
+    (swap) => swap.tokenIn?.id == transaction.swaps[0].tokenIn?.id,
+  );
+
+  const equalTokensOut = transaction.swaps.every(
+    (swap) => swap.tokenOut?.id == transaction.swaps[0].tokenOut?.id,
+  );
+
+  const decimalsTokenIn = transaction.swaps[0].tokenIn?.decimals || 18;
+
+  const totalAmountTokenIn = transaction.swaps.reduce(
+    (acc, swap) =>
+      acc + Number(formatUnits(swap.tokenAmountIn, decimalsTokenIn)),
+    0,
+  );
+
+  return (
+    <>
+      <Table.BodyRow key={transaction.id}>
+        <Table.BodyCell>
+          <button onClick={() => setShowOrdersRows(!showOrdersRows)}>
+            {showOrdersRows ? (
+              <ArrowUpIcon className="h-5 w-5 text-blue9 hover:text-blue10" />
+            ) : (
+              <ArrowDownIcon className="h-5 w-5 text-blue9 hover:text-blue10" />
+            )}
```

Comment:
```suggestion
            {showOrdersRows ? (
              <ChevronUpIcon className="h-5 w-5 text-blue9 hover:text-blue10" />
            ) : (
              <ChevronDownIcon className="h-5 w-5 text-blue9 hover:text-blue10" />
            )}
```

<img width="1241" alt="image" src="https://github.com/bleu-fi/monorepo/assets/63298341/f8823df2-df6b-4605-911f-8733aafb8b2f">